parser_library/src/lsp/lsp_context.cpp

```diff
@@ -106,6 +106,18 @@ void lsp_context::add_opencode(opencode_info_ptr opencode_i, text_data_ref_t tex
     distribute_file_occurences(opencode_->file_occurences);
 }
 
+macro_info_ptr lsp_context::get_macro_info(context::id_index macro_name) const
+{
+    if (!opencode_)
```

Comment:
what are the conditions in which the `opencode_` is empty in this function?

Reply:
It can be empty when parsing is interrupted by cancellation token.
And also when there is a corrupted macro files, so there is no macro definition in it and open code calls the macro twice, the first time we cache the resulting analyzer with no macro definition. The second call reads the cache and tries to get the macro info through this function, but the opencode is not there yet.

Although the second case seems strange, I didn't fix, since it won't happen very often and we issue a diagnostic that the macro file should define a macro.